### PR TITLE
specify data type of websocket message event

### PR DESCRIPTION
**Pull request by jingkaimori**
### Description

Specify data type of `messageEvent.data` in websocket interface.

### Motivation

Specify type so readers can avoid type errors on this interface

### Additional details

related chapter on whatwg: [WebSockets](https://websockets.spec.whatwg.org/#eventdef-websocket-message)

### CODE REVIEW

File: files/en-us/web/api/websocket/message_event/index.md [hunk 1]
```diff
@@ -35,7 +35,11 @@ A {{domxref("MessageEvent")}}. Inherits from {{domxref("Event")}}.
 _In addition to the properties listed below, properties from the parent interface, {{domxref("Event")}}, are available._
 
 - {{domxref("MessageEvent.data", "data")}} {{ReadOnlyInline}}
-  - : The data sent by the message emitter.
+  - : The data sent by the message emitter. This field may be one of types below depending on type of websocket message and {{domxref("WebSocket.binaryType")}}
+    - if received socket message is string type, then this field is a {{jsxref("String")}}.
+    - if received socket message is binary type, then type of this field will be inferred from `binaryType` of this socket.
+      - {{jsxref("ArrayBuffer")}} if `binaryType` is `"arraybuffer"`,
+      - {{domxref("Blob")}} if `binaryType` is `"blob"`.
```
Comment (Josh-Cena) [inline files/en-us/web/api/websocket/message_event/index.md:42]:
```suggestion
  - : The data sent by the message emitter. This field may be one of types below depending on type of websocket message and {{domxref("WebSocket.binaryType")}}.
    - If the received socket message is a string, then this field is a string.
    - If the received socket message is binary type, then the type of this field will be inferred from `binaryType` of this socket:
      - {{jsxref("ArrayBuffer")}} if `binaryType` is `"arraybuffer"`,
      - {{domxref("Blob")}} if `binaryType` is `"blob"`.
```

**Comment by Josh-Cena** [comment]
Pinging @wbamberg for review

**Comment by Josh-Cena** [comment]
(Also, I guess this information should be added to the `data` reference page itself rather than this overview page)

File: files/en-us/web/api/websocket/message_event/index.md [hunk 2]
```diff
@@ -30,7 +30,11 @@ A {{domxref("MessageEvent")}}. Inherits from {{domxref("Event")}}.
 _In addition to the properties listed below, properties from the parent interface, {{domxref("Event")}}, are available._
 
 - {{domxref("MessageEvent.data", "data")}} {{ReadOnlyInline}}
-  - : The data sent by the message emitter.
+  - : The data sent by the message emitter. This field may be one of types below depending on type of websocket message and {{domxref("WebSocket.binaryType")}}.
```
Comment (wbamberg) [inline files/en-us/web/api/websocket/message_event/index.md:33]:
```suggestion
  - : The data sent by the message emitter. The type of this property depends on the type of the WebSocket message and the value of {{domxref("WebSocket.binaryType")}}.
```
...but how do we know the type of the WebSocket message? It seems to be visible as the `opcode` value in https://datatracker.ietf.org/doc/html/rfc6455#section-5.2, but I couldn't see how this is visible in the JS API.

File: files/en-us/web/api/websocket/message_event/index.md [hunk 3]
```diff
@@ -30,7 +30,11 @@ A {{domxref("MessageEvent")}}. Inherits from {{domxref("Event")}}.
...
+    - If the received socket message is a string, then this field is a string.
+    - If the received socket message is binary type, then the type of this field will be inferred from `binaryType` of this socket:
```
Comment (wbamberg) [inline files/en-us/web/api/websocket/message_event/index.md:35]:
```suggestion
    - If the message type is "text", then this field is a string.
    - If the message type is "binary" type, then the type of this property can be inferred from the `binaryType` of this socket:
```

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks for your PR, @jingkaimori !

I had some copy-edit suggestions, but also I wasn't sure how someone can figure out the type of the message, although I can see it in the RFC.

File: files/en-us/web/api/websocket/message_event/index.md [hunk 4]
```diff
@@ -30,7 +30,11 @@ A {{domxref("MessageEvent")}}. Inherits from {{domxref("Event")}}.
 _In addition to the properties listed below, properties from the parent interface, {{domxref("Event")}}, are available._
 
 - {{domxref("MessageEvent.data", "data")}} {{ReadOnlyInline}}
-  - : The data sent by the message emitter.
+  - : The data sent by the message emitter. The type of this property depends on the type of the WebSocket message and the value of {{domxref("WebSocket.binaryType")}}.
```
Comment (Josh-Cena) [inline files/en-us/web/api/websocket/message_event/index.md:33]: Could you move this information to the `data` page? This description here should only be a one-line overview.

File: files/en-us/web/api/websocket/message_event/index.md (hunk 2 above)
Comment (dontcallmedom) [reply files/en-us/web/api/websocket/message_event/index.md]: I'm not sure who is "we" in the above; as a developer, you would know by e.g. testing `typeof data === 'string'` (and if it isn't, then you would check `binaryType` to figure how the binary data is represented). The browser is responsible for mapping the protocol bits to the JS surface.

(in reality, in most cases there is a tight coupling between the JS consumer and the emitter, so the developer may simply know the type from out of band coordination)

Comment (wbamberg) [reply files/en-us/web/api/websocket/message_event/index.md]: Thank you @dontcallmedom , of course, that makes sense.

File: files/en-us/web/api/websocket/message_event/index.md (hunk 4 above)
Comment (wbamberg) [reply files/en-us/web/api/websocket/message_event/index.md:33]: @jingkaimori , could you also make this change please? Then I think we should be able to merge this PR.

**Review by wbamberg — APPROVED** [review]
👍 thank you @jingkaimori !

**Pull request merged**